### PR TITLE
created app shell and misc stubbed views and components

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#2335

*Description of changes:*
- created app shell in App.vue (with Fee Summary component)
- created stub Stepper component
- created stub Actions component
- created stub Define Company component
- created stub Review Confirm component
- updated routes accordingly
- added vue-affix package for Fee Summary component
- added vue2-filters package for Fee Summary component
- temporarily disabled some strict options due to Fee Summary component
- added EntityTypes and FilingCodes
- checked in package-lock.json (needed for GH Actions)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -18,17 +57,42 @@ import { Component, Vue } from 'vue-property-decorator'
...
   }
 })
-export default class App extends Vue {}
-</script>
+export default class App extends Vue {
+  private filingData: Array<FilingDataIF> = []
+  private totalFee: number = 0
+
+  private created (): void {
+    this.filingData.push({ filingTypeCode: 'OTINC', entityType: 'CP' })
```
Comment (cameron-freshworks) [inline src/App.vue]: Should we start filingCodes and entityTypes ENUMs for these property values?

File: src/components/common/EntityInfo.vue [hunk 2]
```diff
@@ -54,7 +54,6 @@ export default class EntityInfo extends Vue {
 }
 </script>
 
-<!-- eslint-disable max-len -->
```
Comment (cameron-freshworks) [inline src/components/common/EntityInfo.vue]: Thanks!

File: src/components/common/Stepper.vue [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  </v-container>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Watch } from 'vue-property-decorator'
+
+@Component
+export default class Stepper extends Vue {
+  @Watch('$route', { immediate: true })
```
Comment (cameron-freshworks) [inline src/components/common/Stepper.vue]: What does the option `immediate: true` do?

File: src/interfaces/index.ts [hunk 4]
```diff
@@ -7,3 +7,6 @@ export * from './store-interfaces/state-interface'
 export * from './store-interfaces/action-interface'
 export * from './store-interfaces/tombstone-interface'
 export * from './store-interfaces/getter-interface'
+
+// Others
+export * from './filingData-interface'
```
Comment (cameron-freshworks) [inline src/interfaces/index.ts]: I like it! As far as directory structure ( thinking ahead ), would this fall into any interface type category that we might add too?

File: src/views/DefineCompany.vue [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      </section>
+    </v-card>
+  </div>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Vue } from 'vue-property-decorator'
+
+@Component({})
```
Comment (cameron-freshworks) [inline src/views/DefineCompany.vue]: I know we were used to putting in the `({options})` but it isn't required. Is this how you would like to see the decorator done?

File: src/views/ReviewConfirm.vue [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      </section>
+    </v-card>
+  </div>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Vue } from 'vue-property-decorator'
+
+@Component({})
```
Comment (cameron-freshworks) [inline src/views/ReviewConfirm.vue]: Same as above, just curious if you would prefer this!

File: tsconfig.json [hunk 7]
```diff
@@ -2,7 +2,13 @@
   "compilerOptions": {
     "target": "esnext",
     "module": "esnext",
-    "strict": true,
```
Comment (cameron-freshworks) [inline tsconfig.json]: Why the change here? Is there a specific property that you didn't want strict that was covered in the list?

File: src/App.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/App.vue]: Sure, I can add those.

Comment (severinbeauvais) [reply src/App.vue]: Done.

File: src/components/common/Stepper.vue (hunk 3 above)
Comment (severinbeauvais) [reply src/components/common/Stepper.vue]: You get an initial event (before the value changes).

There's also an option `deep: boolean` that generates an event when an object's properties change.

File: src/interfaces/index.ts (hunk 4 above)
Comment (severinbeauvais) [reply src/interfaces/index.ts]: I'm not sure yet, which is why I didn't create one :)

I expect this will either be clearer in the future or not matter :)

File: src/views/DefineCompany.vue (hunk 5 above)
Comment (severinbeauvais) [reply src/views/DefineCompany.vue]: Done like this. At one point I got lint errors when I didn't specify the `{}`.

File: src/views/ReviewConfirm.vue (hunk 6 above)
Comment (severinbeauvais) [reply src/views/ReviewConfirm.vue]: I don't think I have a preference.

File: tsconfig.json (hunk 7 above)
Comment (severinbeauvais) [reply tsconfig.json]: As you know, `strict` covers all the other properties. I needed to disable a couple of properties specifically (due to SbcFeeSummary) but I also wanted to keep the other checks enabled.

File: src/App.vue [hunk 8]
```diff
@@ -78,7 +81,7 @@ export default class App extends Vue {
   private totalFee: number = 0
 
   private created (): void {
-    this.filingData.push({ filingTypeCode: 'OTINC', entityType: 'CP' })
+    this.filingData.push({ filingTypeCode: FilingCodes.INCORPORATION_BC, entityType: EntityTypes.BCOMP })
```
Comment (cameron-freshworks) [inline src/App.vue]: Right on! 👍

File: src/components/common/Stepper.vue (hunk 3 above)
Comment (cameron-freshworks) [reply src/components/common/Stepper.vue]: interesting, thanks!

File: src/interfaces/index.ts (hunk 4 above)
Comment (cameron-freshworks) [reply src/interfaces/index.ts]: Sounds good, true agile ;)

File: tsconfig.json (hunk 7 above)
Comment (cameron-freshworks) [reply tsconfig.json]: If you feel this doesn't compromise our goal of being strict and won't carry over into other elements of development, i'm 100% fine with it. Just at first glance, those 3 options certainly do lessen the rules. Is this the only way?

Comment (cameron-freshworks) [reply tsconfig.json]: Discussed with Severin that this is a temporary change until we can fix some type issues with sbc common components. Resolved!

Comment (severinbeauvais) [reply tsconfig.json]: I have created ticket /bcgov/entity#2357 to get the SbcFeeSummary updated so that we can go back to `strict: true`.
